tests/ImageSharp.Tests/Formats/WebP/WebpEncoderTests.cs

```diff
@@ -101,6 +101,24 @@ public void Encode_Lossless_WithDifferentMethodAndQuality_Works<TPixel>(TestImag
             image.VerifyEncoder(provider, "webp", testOutputDetails, encoder);
         }
 
+        [Theory]
+        [WithFile(Lossy.NoFilter06, PixelTypes.Rgba32, 15114)]
+        public void Encode_Lossless_WithBestQuality_HasExpectedSize<TPixel>(TestImageProvider<TPixel> provider, int expectedBytes)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            var encoder = new WebpEncoder()
+            {
+                FileFormat = WebpFileFormatType.Lossless,
+                Method = WebpEncodingMethod.BestQuality
+            };
+
+            using Image<TPixel> image = provider.GetImage();
+            using var memoryStream = new MemoryStream();
+            image.Save(memoryStream, encoder);
+
+            Assert.Equal(memoryStream.Length, expectedBytes);
```

Comment:
Do we need strict equality? Doesn't the spec allow some space for the implementation?

Reply:
No the spec is not that strict. The analyses step could have choosen different transformation, e.g. here the analyses step has choosen to use the following features PREDICTION CROSS-COLOR-TRANSFORM SUBTRACT-GREEN. If one of those transformations would not been applied, the image would be still valid, but a bit larger.

Nonetheless i think we should try to aim to be as close to the reference implementation as possible and this is what libwebp would have encoded to.

Reply:
Ok then, we can relax this restriction later if it will become necessary for some reason in the future.